planner/core/rule_decorrelate.go

```diff
@@ -214,7 +215,7 @@ func (s *decorrelateSolver) optimize(ctx context.Context, p LogicalPlan) (Logica
 								agg.schema.Columns[agg.schema.Len()-1].RetType = newFunc.RetTp
 							}
 							// If group by cols don't contain the join key, add it into this.
-							if agg.getGbyColIndex(eqCond.GetArgs()[1].(*expression.Column)) == -1 {
+							if !groupByColumns.Contains(eqCond.GetArgs()[1].(*expression.Column)) {
```

Comment:
Would it append same group by columns since it is based on old columns?

Reply:
In fact, the `la.groupByCols` is not changed in this for-loop. `agg.getGbyColIndex` is also based on old columns. If it can append same group by columns, it would happen before this change.

Reply:
Yes, I know it is the same. I just raise this question and may be we need to take care of it.

Reply:
Sure, I'm gonna check this.

Reply:
Yeah, this could happen. I have updated the PR description, and fixed it. PTAL.